agreement/pseudonode.go

```diff
@@ -438,15 +447,42 @@ func (t pseudonodeVotesTask) execute(verifier *AsyncVoteVerifier, quit chan stru
 	t.node.monitor.dec(pseudonodeCoserviceType)
 
 	// push results into channel.
+	var outputTimeout <-chan time.Time
+	var timedOut bool
+
+verifiedVotesLoop:
 	for _, r := range verifiedResults {
 		select {
 		case t.out <- messageEvent{T: voteVerified, Input: r.message, Err: makeSerErr(r.err)}:
 			t.node.keys.Record(r.v.R.Sender, r.v.R.Round, account.Vote)
+			continue verifiedVotesLoop
 		case <-quit:
 			return
 		case <-t.context.Done():
 			// we done care about the output anymore; just exit.
 			return
+		default:
+		}
+
+		if outputTimeout == nil && !timedOut {
+			outputTimeout = time.After(maxPseudonodeOutputWaitDuration)
```

Comment:
So this is to save the extra resources required to make a timeout channel per iteration, is that right? I wonder how often you are going to end up breaking out into the default and retry ... it seems like you might end up here most of the time, but hard to say.

Reply:
It's timing-dependent, but it would allocate a timer only once.

Reply:
I was just thinking, maybe it is not worth the added complexity / code duplication if you are almost always going to end up calling time.After() on every iteration. But without evidence that is true I wouldn't know